experimental/Schemes/duValSing.jl

```diff
@@ -0,0 +1,262 @@
+@doc raw"""
+    has_du_val_singularities(X::Scheme)
+
+Return whether the given `X` has at most du Val (surface) singularities.
+
+# Example:
+```jldoctest
+julia> R,(x,y,z,w) = QQ["x","y","z","w"]
+(Multivariate polynomial ring in 4 variables over QQ, QQMPolyRingElem[x, y, z, w])
+
+julia> I = ideal(R,[w,x^2+y^3+z^4])
+ideal(w, x^2 + y^3 + z^4)
+
+julia> Rq,_ = quo(R,I)
+(Quotient of multivariate polynomial ring by ideal with 2 generators, Map from
+Multivariate polynomial ring in 4 variables over QQ to Rq defined by a julia-function with inverse)
+
+julia> X = Spec(Rq)
+Spec of Quotient of multivariate polynomial ring by ideal with 2 generators
+
+julia> has_du_val_singularities(X)
+true
+
+```
+"""
+function has_du_val_singularities(X::AbsProjectiveScheme{<:Field,<:Any})
+  return has_du_val_singularities(covered_scheme(X))
+end
+
+function has_du_val_singularities(X::AbsSpec{<:Field,<:Any})
+  R = OO(X)
+  I = modulus(R)
+  
+  J = image_ideal(singular_locus(X)[2])
+  J = ideal(base_ring(I), lift.(gens(J))) + I
+  dim(J) == 0 || return false                            ## non-isolated
+  return is_du_val_singularity(X,J)
+end
+
+function has_du_val_singularities(X::AbsCoveredScheme{<:Field})
+  C = (has_attribute(X, :simplified_covering) ? simplified_covering(X) : default_covering(X))
+
+  I = ideal_sheaf_of_singular_locus(X)
+  decomp = maximal_associated_points(I)
+
+  ## we do the double loop here to avoid unnecessary checks
+  for J in decomp
+    for U in C
+      !isone(J(U)) || continue
+      is_du_val_singularity(U,saturated_ideal(J(U))) || return false    ## testing the point in one chart suffices
+      break
+    end
+  end
+
+  return true
+end
+
+@doc raw"""
+    is_du_val_singularity(X::AbsSpec, I::Ideal)
+
+Return whether the given X has at most du Val (surface) singularities at the geometric points
+specified by the ideal I.
+
+**Note**: For the ideal I in a ring R, dim(R/I) = 0 is asserted
+
+# Example:
+```jldoctest
+julia> R,(x,y,z,w) = QQ["x","y","z","w"]
+(Multivariate polynomial ring in 4 variables over QQ, QQMPolyRingElem[x, y, z, w])
+
+julia> I = ideal(R,[w,x^2+y^3+z^4])
+ideal(w, x^2 + y^3 + z^4)
+
+julia> Rq,_ = quo(R,I)
+(Quotient of multivariate polynomial ring by ideal with 2 generators, Map from
+Multivariate polynomial ring in 4 variables over QQ to Rq defined by a julia-function with inverse)
+
+julia> J = ideal(R,[x,y,z,w])
+ideal(x, y, z, w)
+
+julia> X = Spec(Rq)
+Spec of Quotient of multivariate polynomial ring by ideal with 2 generators
+
+julia> is_du_val_singularity(X,J)
+true
+
+```
+"""
+function is_du_val_singularity(X::AbsSpec{<:Field,<:Any},I::Ideal)
+  OOX = OO(X)
+  dim(X) == 2 || error("Scheme not of dimension 2")
+  J = modulus(OOX)
+  !isone(J) || error("Scheme is empty")
+  !iszero(J) || return true                  ## X smooth
+
+  R = base_ring(I)
+  kk = base_ring(R)
+  characteristic(kk) == 0 || error("only available in characteristic zero")
```

Comment:
Can we do characteristic $p\geq 5$ ,say?

Reply:
According to the classification of Greuel and Kröning, the classification differs for p \in \{2,3,5\}. Therefore I put in >5 and this should do the job.
Sideremark: duVal-Singularities are only called duVal-Singularities in characteristic zero -- but I understand your need to pass to large positive characteristic to avoid intermediate coefficient swell.